### ISSUE TITLE
NullInjectorError: No provider for InjectionToken I18NEXT_SERVICE!

### ISSUE THREAD

**Issue by httpete** [Romanchuk/angular-i18next#12]
Hi Romanchuk,

Any clues? I am on the latest Angular 5.2.2. I am trying to use i18next from a shared library - I wonder if that has anything to do with it?

Pete

**Comment by Romanchuk** [comment]
@httpete Hi, dont miss I18NextModule.forRoot() method? Or may be you call translation methods before forRoot was called.
Also this article could help:
https://www.gurustop.net/blog/2017/02/14/shared-modules-in-angular-apps-providers-best-practices-and-what-does-forroot-do/

Did issue appear right after angular update?

**Comment by httpete** [comment]
Hi, I found that my webpack config needed this in my resolve, the part about "mainFields" lit it up. I guess you use a different main field in package.json?

```
        resolve: {
            extensions: ['.ts', '.js'],
            modules: [
                "./node_modules",
                "./node_modules"
            ],
            symlinks: false,
            mainFields: [
                "browser",
                "module",
                "main"
            ],
            alias: {
                angular: path.resolve(path.join(__dirname, 'node_modules', 'angular')),
                lodash: path.resolve(path.join(__dirname, 'node_modules', 'lodash-amd')),
                jquery: "jquery/dist/jquery.slim.js"
            }
        },
```

**Comment by Romanchuk** [comment]
In angular-i18next package.json i use 'module'. This is proposal for nodejs.
https://stackoverflow.com/questions/42708484/what-is-the-module-package-json-field-for

May be i should also specify 'main' parameter because 'module' is not official part of package.json
https://docs.npmjs.com/files/package.json#main

Here is angular-i18next package.json
```json
{
    "name": "angular-i18next",
    "version": "3.2.0",
    "homepage": "https://github.com/Romanchuk/angular-i18next#readme",
    "author": {
        "name": "Sergey Romanchuk"
    },
    "license": "MIT",
    "description": "i18next module for Angular",
    "repository": {
        "type": "git",
        "url": "git+https://github.com/Romanchuk/angular-i18next.git"
    },
    "keywords": [
        "i18n",
        "i18next",
        "angular"
    ],
    "bugs": {
        "url": "https://github.com/Romanchuk/angular-i18next/issues"
    },
    "maintainers": [{
        "email": "rezety@gmail.com",
        "name": "Sergey Romanchuk"
    }],
    "module": "angular-i18next.es5.js",
    "es2015": "angular-i18next.js",
    "typings": "angular-i18next.d.ts",
    "peerDependencies": {
        "@angular/common": ">=4.0.0 <6.0.0",
        "@angular/core": ">=4.0.0 <6.0.0",
        "@angular/platform-browser": ">=4.0.0 <6.0.0",
        "rxjs": "^5.5.2",
        "i18next": ">=8.4.0 <11.0.0"
    }
}
```

**Comment by httpete** [comment]
main is what you want I am certain.
    On Saturday, February 3, 2018, 8:07:40 AM EST, Sergey Romanchuk <notifications@github.com> wrote:  
 
 
In angular-i18next package.json i use 'module'. This is proposal for nodejs.
https://stackoverflow.com/questions/42708484/what-is-the-module-package-json-field-for

May be i should also specify 'main' parameter because 'module' is not official part of package.json
https://docs.npmjs.com/files/package.json#main

Here is angular-i18next package.json
{
    "name": "angular-i18next",
    "version": "3.2.0",
    "homepage": "https://github.com/Romanchuk/angular-i18next#readme",
    "author": {
        "name": "Sergey Romanchuk"
    },
    "license": "MIT",
    "description": "i18next module for Angular",
    "repository": {
        "type": "git",
        "url": "git+https://github.com/Romanchuk/angular-i18next.git"
    },
    "keywords": [
        "i18n",
        "i18next",
        "angular"
    ],
    "bugs": {
        "url": "https://github.com/Romanchuk/angular-i18next/issues"
    },
    "maintainers": [{
        "email": "rezety@gmail.com",
        "name": "Sergey Romanchuk"
    }],
    "module": "angular-i18next.es5.js",
    "es2015": "angular-i18next.js",
    "typings": "angular-i18next.d.ts",
    "peerDependencies": {
        "@angular/common": ">=4.0.0 <6.0.0",
        "@angular/core": ">=4.0.0 <6.0.0",
        "@angular/platform-browser": ">=4.0.0 <6.0.0",
        "rxjs": "^5.5.2",
        "i18next": ">=8.4.0 <11.0.0"
    }
}

**Comment by Romanchuk** [comment]
@httpete 
I found great [article](http://2ality.com/2017/04/setting-up-multi-platform-packages.html) about setting up packages.

Is says:
> Angular v4 delivers each package in three formats:
> 
> UMD: via property main
> ES5/ESM: via property module
> ES6/ESM: via property es2015
> 
> This is what its package.json looks like:
> ```
> {
>   "name": "@angular/core",
>   "main": "./bundles/core.umd.js",
>   "module": "./@angular/core.es5.js",
>   "es2015": "./@angular/core.js",
>   ···
> }
> ```

I'm going to add "main" property with umd compiled module ASAP.

But anyway i still can't reproduce this issue using webpack ([demo app](https://github.com/Romanchuk/angular-i18nex-demo)) neither using angular-cli. I never used to set "mainFields" in resolvers. Webpack [has defaults](http://2ality.com/2017/04/setting-up-multi-platform-packages.html#webpack) that are always include 'module' mainField

**Comment by Romanchuk** [comment]
@httpete just published 3.3.0-beta.1 version with 'main' prop setted to umd compiled file.
Please try it
 `npm install angular-i18next@beta --save`

**Comment by httpete** [comment]
Romanchuk, interestingly it didn't change anything, until I changed my tsconfig.json: this might have to do with the fact that I am running on file:../ links for local development of my shared library.


`{
  "compileOnSave": false,
  "compilerOptions": {
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": false,
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "target": "es5",
    "paths": {
      "@angular/*": [
        "../node_modules/@angular/*"
      ],
      "@ng-bootstrap/*": [
        "../node_modules/@ng-bootstrap/*"
      ],
      "@uirouter/*": [
        "../node_modules/@uirouter/*"
      ],
      "angular-i18next": [
        "../node_modules/angular-i18next"
      ]
    },
    "lib": [
      "es2017",
      "dom"
    ]
  }
}
`
